### DISCUSSION TITLE
Validation of Configuration and Code for Splice Usage in Xray-core

### DISCUSSION THREAD

**Discussion by us254** [XTLS/Xray-core#2570] [Q&A]
Is the following configuration true based on the provided code and commit?

```
{
  "outbounds": [
    {
      "protocol": "freedom",
      "settings": {
        "domainStrategy": "UseIP",
        "splice": true
      },
      "tag": "freedom"
    }
  ]
}
```
https://github.com/XTLS/Xray-core/commit/efd32b0fb2a4f1b2d2e98b21916cfd9aa0e0b497
```
outbound.Name = "freedom"
inbound := session.InboundFromContext(ctx)
if inbound != nil {
    inbound.SetCanSpliceCopy(1)
}
```

**Comment by yuhan6665** [comment]
You can try set environment variable `xray.buf.splice` to `enable` for now
@RPRX has some concern it might cause some issue so it is not enabled by default

**Comment by us254** [comment]
So, does it mean it should look like this?

1. Open your terminal.

2. Use the `nano` text editor to edit the `~/.bashrc` file:
   ```
   nano ~/.bashrc
   ```

3. Add the following line to the `~/.bashrc` file, setting the value to "enable":
   ```
   export xray_buf_splice=enable
   
   ```
4. Save the changes and exit the text editor. In `nano`, you can do this by pressing `Ctrl` + `O` to write the file and then `Enter`, and then `Ctrl` + `X` to exit.

5. To apply the changes, you can either restart your terminal or run the following command to reload the `~/.bashrc` file:
   ```
   source ~/.bashrc
   ```

**Comment by yuhan6665** [comment]
Yeah you can search for "splice" in logs to confirm it is working. (Where you use freedom outbound, usually a Linux server)

**Comment by us254** [comment]
I noticed an interesting entry in my log. The `CopyRawConn splice` appeared, indicating that the splice operation was successful.

**Comment by chika0801** [comment]
I added a new `Environment="XRAY_BUF_SPLICE=enable"` to the xray.service file on the server side. I set the logging level to debug. i observed a change in the logging .

```
[Unit]
Description=Xray Service
Documentation=https://github.com/xtls
After=network.target nss-lookup.target

[Service]
User=nobody
CapabilityBoundingSet=CAP_NET_ADMIN CAP_NET_BIND_SERVICE
AmbientCapabilities=CAP_NET_ADMIN CAP_NET_BIND_SERVICE
NoNewPrivileges=true
ExecStart=/usr/local/bin/xray run -config /usr/local/etc/xray/config.json
Restart=on-failure
RestartPreventExitStatus=23
LimitNPROC=10000
LimitNOFILE=1000000
Environment="XRAY_BUF_SPLICE=enable"

[Install]
WantedBy=multi-user.target

```

```
2023/09/20 16:28:29 [Info] [361353812] proxy: XtlsFilterTls found tls 1.3! 4821 TLS_AES_128_GCM_SHA256
2023/09/20 16:28:29 [Info] [361353812] proxy: XtlsPadding 4821 253 0
2023/09/20 16:28:29 [Info] [361353812] proxy: Xtls Unpadding new block, content 64 padding 859 command 0
2023/09/20 16:28:29 [Info] [361353812] proxy: XtlsPadding 614 497 2
2023/09/20 16:28:29 [Info] [361353812] proxy: Xtls Unpadding new block, content 243 padding 846 command 2
2023/09/20 16:28:29 [Info] [361353812] proxy: CopyRawConn readv
2023/09/20 16:28:29 [Info] [361353812] proxy: CopyRawConn splice
```

thanks

**Comment by chika0801** [comment]
My geekbench5 1 core score for this VPS is 500. with this turned on, I tested using speedtest.net. During the downlink speed test, the CPU usage of the VPS went from 99% before to 60% now. The downstream speed was 600Mbps before due to the bottleneck caused by the CPU, now it can reach 700Mbps. the `SPLICE `parameter is fantastic.

**Reply by yuhan6665** [reply]
it is on by default now
